404 popup when disconnecting

After connecting to a target and going to the Recordings tab, then disconnecting by choosing "Select Target..." from the dropdown. I reliably get a "Request failed (404 Not Found)" popup and the following in the Container JFR log:

```
io.vertx.ext.web.handler.impl.HttpStatusException: Not Found
Caused by: org.openjdk.jmc.rjmx.ConnectionException caused by java.io.IOException: Failed to retrieve RMIServer stub: javax.naming.ConfigurationException [Root exception is java.rmi.UnknownHostException: Unknown host: undefined; nested exception is:
	java.net.UnknownHostException: undefined]
	at org.openjdk.jmc.rjmx.internal.RJMXConnection.connect(RJMXConnection.java:441)
	at com.redhat.rhjmc.containerjfr.core.net.JFRConnection.attemptConnect(JFRConnection.java:212)
	at com.redhat.rhjmc.containerjfr.core.net.JFRConnection.connect(JFRConnection.java:169)
	at com.redhat.rhjmc.containerjfr.core.net.JFRConnection.getService(JFRConnection.java:109)
	at com.redhat.rhjmc.containerjfr.net.web.handlers.TargetRecordingsGetHandler.lambda$handleAuthenticated$0(TargetRecordingsGetHandler.java:103)
	at com.redhat.rhjmc.containerjfr.net.TargetConnectionManager.executeConnectedTask(TargetConnectionManager.java:87)
	at com.redhat.rhjmc.containerjfr.net.web.handlers.TargetRecordingsGetHandler.handleAuthenticated(TargetRecordingsGetHandler.java:99)
	at com.redhat.rhjmc.containerjfr.net.web.handlers.AbstractAuthenticatedRequestHandler.handle(AbstractAuthenticatedRequestHandler.java:82)
	at com.redhat.rhjmc.containerjfr.net.web.handlers.AbstractAuthenticatedRequestHandler.handle(AbstractAuthenticatedRequestHandler.java:61)
	at io.vertx.ext.web.impl.BlockingHandlerDecorator.lambda$handle$0(BlockingHandlerDecorator.java:48)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:313)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.IOException: Failed to retrieve RMIServer stub: javax.naming.ConfigurationException [Root exception is java.rmi.UnknownHostException: Unknown host: undefined; nested exception is:
	java.net.UnknownHostException: undefined]
	at java.management.rmi/javax.management.remote.rmi.RMIConnector.connect(RMIConnector.java:370)
	at org.openjdk.jmc.rjmx.internal.RJMXConnection.connectJmxConnector(RJMXConnection.java:487)
	at org.openjdk.jmc.rjmx.internal.RJMXConnection.establishConnection(RJMXConnection.java:464)
	at org.openjdk.jmc.rjmx.internal.RJMXConnection.connect(RJMXConnection.java:434)
	... 15 more
Caused by: javax.naming.ConfigurationException [Root exception is java.rmi.UnknownHostException: Unknown host: undefined; nested exception is:
	java.net.UnknownHostException: undefined]
	at jdk.naming.rmi/com.sun.jndi.rmi.registry.RegistryContext.lookup(RegistryContext.java:137)
	at java.naming/com.sun.jndi.toolkit.url.GenericURLContext.lookup(GenericURLContext.java:207)
	at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
	at java.management.rmi/javax.management.remote.rmi.RMIConnector.findRMIServerJNDI(RMIConnector.java:1839)
	at java.management.rmi/javax.management.remote.rmi.RMIConnector.findRMIServer(RMIConnector.java:1813)
	at java.management.rmi/javax.management.remote.rmi.RMIConnector.connect(RMIConnector.java:302)
	... 18 more
Caused by: java.rmi.UnknownHostException: Unknown host: undefined; nested exception is:
	java.net.UnknownHostException: undefined
	at java.rmi/sun.rmi.transport.tcp.TCPEndpoint.newSocket(TCPEndpoint.java:620)
	at java.rmi/sun.rmi.transport.tcp.TCPChannel.createConnection(TCPChannel.java:209)
	at java.rmi/sun.rmi.transport.tcp.TCPChannel.newConnection(TCPChannel.java:196)
	at java.rmi/sun.rmi.server.UnicastRef.newCall(UnicastRef.java:343)
	at java.rmi/sun.rmi.registry.RegistryImpl_Stub.lookup(RegistryImpl_Stub.java:116)
	at jdk.naming.rmi/com.sun.jndi.rmi.registry.RegistryContext.lookup(RegistryContext.java:133)
	... 23 more
Caused by: java.net.UnknownHostException: undefined
	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:220)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:403)
	at java.base/java.net.Socket.connect(Socket.java:609)
	at java.base/java.net.Socket.connect(Socket.java:558)
	at java.base/java.net.Socket.<init>(Socket.java:454)
	at java.base/java.net.Socket.<init>(Socket.java:231)
	at java.rmi/sun.rmi.transport.tcp.TCPDirectSocketFactory.createSocket(TCPDirectSocketFactory.java:40)
	at java.rmi/sun.rmi.transport.tcp.TCPEndpoint.newSocket(TCPEndpoint.java:617)
	... 28 more
```

This was with 1.0.0-BETA4 deployed by the operator. Not sure if this is an issue in main.

--- a/src/app/Events/EventTemplates.tsx
+++ b/src/app/Events/EventTemplates.tsx
@@ -38,12 +38,13 @@
 import * as React from 'react';
 import { ServiceContext } from '@app/Shared/Services/Services';
 import { EventTemplate } from '@app/Shared/Services/Api.service';
+import {NO_TARGET} from '@app/Shared/Services/Target.service';
 import { useSubscriptions } from '@app/utils/useSubscriptions';
 import { ActionGroup, Button, FileUpload, Form, FormGroup, Modal, ModalVariant, Toolbar, ToolbarContent, ToolbarGroup, ToolbarItem, TextInput } from '@patternfly/react-core';
 import { PlusIcon } from '@patternfly/react-icons';
 import { Table, TableBody, TableHeader, TableVariant, IAction, IRowData, IExtraData, ISortBy, SortByDirection, sortable } from '@patternfly/react-table';
 import { useHistory } from 'react-router-dom';
-import { concatMap, first } from 'rxjs/operators';
+import { concatMap, filter, first } from 'rxjs/operators';
 import { LoadingView } from '@app/LoadingView/LoadingView';
 import { ErrorView } from '@app/ErrorView/ErrorView';
 
@@ -105,6 +106,7 @@ export const EventTemplates = () => {
     addSubscription(
       context.target.target()
       .pipe(
+        filter(target => target !== NO_TARGET),
         first(),
         concatMap(target => context.api.doGet<EventTemplate[]>(`targets/${encodeURIComponent(target.connectUrl)}/templates`)),
       ).subscribe(value => handleTemplates(value), err => handleError(err))

--- a/src/app/Events/EventTypes.tsx
+++ b/src/app/Events/EventTypes.tsx
@@ -37,10 +37,11 @@
  */
 import * as React from 'react';
 import { ServiceContext } from '@app/Shared/Services/Services';
+import {NO_TARGET} from '@app/Shared/Services/Target.service';
 import { useSubscriptions } from '@app/utils/useSubscriptions';
 import { Toolbar, ToolbarContent, ToolbarItem, ToolbarItemVariant, Pagination, TextInput } from '@patternfly/react-core';
 import { expandable, Table, TableBody, TableHeader, TableVariant } from '@patternfly/react-table';
-import { concatMap, first } from 'rxjs/operators';
+import { concatMap, filter, first } from 'rxjs/operators';
 import { LoadingView } from '@app/LoadingView/LoadingView';
 import { ErrorView } from '@app/ErrorView/ErrorView';
 
@@ -105,6 +106,7 @@ export const EventTypes = () => {
     addSubscription(
       context.target.target()
         .pipe(
+          filter(target => target !== NO_TARGET),
           first(),
           concatMap(target => context.api.doGet<EventType[]>(`targets/${encodeURIComponent(target.connectUrl)}/events`)),
         )

--- a/src/app/RecordingList/ActiveRecordingsList.tsx
+++ b/src/app/RecordingList/ActiveRecordingsList.tsx
@@ -36,17 +36,18 @@
  * SOFTWARE.
  */
 
+import {NotificationsContext} from '@app/Notifications/Notifications';
+import {Recording, RecordingState} from '@app/Shared/Services/Api.service';
+import {ServiceContext} from '@app/Shared/Services/Services';
+import {NO_TARGET} from '@app/Shared/Services/Target.service';
+import {useSubscriptions} from '@app/utils/useSubscriptions';
+import {Button, DataListAction, DataListCell, DataListCheck, DataListContent, DataListItem, DataListItemCells, DataListItemRow, DataListToggle, Dropdown, DropdownItem, DropdownPosition, KebabToggle, Text, Toolbar, ToolbarContent, ToolbarItem} from '@patternfly/react-core';
 import * as React from 'react';
-import { NotificationsContext } from '@app/Notifications/Notifications';
-import { Recording, RecordingState } from '@app/Shared/Services/Api.service';
-import { ServiceContext } from '@app/Shared/Services/Services';
-import { useSubscriptions } from '@app/utils/useSubscriptions';
-import { Button, DataListAction, DataListCell, DataListCheck, DataListContent, DataListItem, DataListItemCells, DataListItemRow, DataListToggle, Dropdown, DropdownItem, DropdownPosition, KebabToggle, Text, Toolbar, ToolbarContent, ToolbarItem } from '@patternfly/react-core';
-import { useHistory, useRouteMatch } from 'react-router-dom';
-import { forkJoin, Observable } from 'rxjs';
-import { concatMap, first, tap } from 'rxjs/operators';
-import { RecordingsDataTable } from './RecordingsDataTable';
-import { ReportFrame } from './ReportFrame';
+import {useHistory, useRouteMatch} from 'react-router-dom';
+import {forkJoin, Observable} from 'rxjs';
+import {concatMap, filter, first} from 'rxjs/operators';
+import {RecordingsDataTable} from './RecordingsDataTable';
+import {ReportFrame} from './ReportFrame';
 
 export interface ActiveRecordingsListProps {
   archiveEnabled: boolean;
@@ -108,6 +109,7 @@ export const ActiveRecordingsList: React.FunctionComponent<ActiveRecordingsListP
     addSubscription(
       context.target.target()
       .pipe(
+        filter(target => target !== NO_TARGET),
         concatMap(target => context.api.doGet<Recording[]>(`targets/${encodeURIComponent(target.connectUrl)}/recordings`)),
         first(),
       ).subscribe(value => handleRecordings(value), err => handleError(err))

